api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -117,20 +117,26 @@ public void checkInitialCreditsUsageForUsers(
             .map(l -> Long.toString(l))
             .collect(Collectors.joining(","));
     logger.info(String.format("Checking billing usage for user IDs: %s ", userIdsAsString));
-    // Current cost in DB
-    List<WorkspaceCostView> allCostsInDbForUsers = getAllCostsInDbForUsers(users);
+    // Current cost in DB for those workspaces which have not been "recently" updated, as defined by
+    // a config value.  This guards against excessive updating of workspaces.
+    List<WorkspaceCostView> dbCostsForNotRecentlyUpdatedWorkspaces =
```

Comment:
I was confused about the "no workspaces need updates" logic until I realized that this is not actually *all* costs.  I renamed this and added comments to match my new understanding.